Develop customer new v3.3

charlie-ai opened the pull request:
#### Testing
Verified by CI

haiwentelink committed:
riscv:telink: update commit id and nvs config.

- update commit id for zephyr.
- add NVS_RAM_OPTIMIZATION option.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>

haiwentelink committed:
[Telink] ieee802154 optimization for power consumption (#360)

* [Telink] ieee802154 optimization for power consumption

- reduce COMMON_LIBC_MALLOC_ARENA_SIZE to 13.5KB.
- add configs
  - OPENTHREAD_CSMABACKOFF_OPTIMIZATION,
  - ZEPHYR_NETBUFFER_OPTIMIZATION,
  - IEEE802154_TLX_OPTIMIZATION.
- reduce PACKETBUFFER_CAPACITY_MAX to 1024.
- remove shell in tl321x light switch app CI.
- update zephyr revision.
- set csma backoff in retention mode .

Signed-off-by: Damien Ji <yinghao.ji@telink-semi.com>
Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
Signed-off-by: Lingzhi Chen <lingzhi.chen@telink-semi.com>
Signed-off-by: Liang Zhong <liang.zhong@telink-semi.com>

s07641069 commented on config/telink/chip-module/Kconfig.defaults:
```diff
@@ -262,9 +263,21 @@ config TELINK_SOC_REBOOT_ON_FAULT
 config NVS_LOOKUP_CACHE_SIZE
-    default 400  if SOC_RISCV_TELINK_TL321X
-    default 2048 if !PM
+    default 2048  if SOC_RISCV_TELINK_TL321X
+    default 2048  if !PM
+    default 2048
```
In case if all values are same we can use only one default value instead of three:
`default 2048`

s07641069 commented on src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp:
```diff
@@ -66,6 +66,10 @@ extern "C" void otSysProcessDrivers(otInstance * aInstance);
 extern "C" void otAppCliInit(otInstance * aInstance);
 #endif
 
+#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION && CONFIG_IEEE802154_TLX_OPTIMIZATION
```
I think better to use `#if defined` for both configs like:
`#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION && defined CONFIG_IEEE802154_TLX_OPTIMIZATION`
because in case if `CONFIG_IEEE802154_TLX_OPTIMIZATION=n` it will be undefined

charlie-ai replied:
i agree

s07641069 commented on src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp:
```diff
@@ -379,6 +383,14 @@ bool GenericThreadStackManagerImpl_OpenThread<ImplClass>::_IsThreadAttached()
     curRole = otThreadGetDeviceRole(mOTInst);
     Impl()->UnlockThreadStack();
 
+#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION && CONFIG_IEEE802154_TLX_OPTIMIZATION
```
same here
`#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION && defined CONFIG_IEEE802154_TLX_OPTIMIZATION`

charlie-ai replied:
i agree

s07641069 commented:
Also I suggest you to include changes from https://github.com/project-chip/connectedhomeip/pull/36691 which add Matter GitHub CI builds for both Zephyr versions or wait till this PR will be merged into master and after you can rebase your branch.
It will help you to check that all targets builds fine for both Zephyr versions.

fengtai-telink commented:
> Also I suggest you to include changes from [project-chip#36691](https://github.com/project-chip/connectedhomeip/pull/36691) which add Matter GitHub CI builds for both Zephyr versions or wait till this PR will be merged into master and after you can rebase your branch. It will help you to check that all targets builds fine for both Zephyr versions.

Hi @s07641069 @haiwentelink  ,
The current branch(**develop_customer**) is a separate PM development version provided to users based on Zephyr_3.3 . Currently, Zephyr_3.7 will not be integrated . And than it only supports tl3218x and b92 . Therefore, the CI only requires tl3218x and b92 series about PM . 
@charlie-ai will further improve the CI targets of tl3218x and b92 . Thank ~

damien0x0023 commented on src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp:
```diff
@@ -66,6 +66,10 @@ extern "C" void otSysProcessDrivers(otInstance * aInstance);
 extern "C" void otAppCliInit(otInstance * aInstance);
 #endif
 
+#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION
```
#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION && CONFIG_IEEE802154_TLX_OPTIMIZATION

damien0x0023 commented on src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp:
```diff
@@ -379,6 +383,14 @@ bool GenericThreadStackManagerImpl_OpenThread<ImplClass>::_IsThreadAttached()
     curRole = otThreadGetDeviceRole(mOTInst);
     Impl()->UnlockThreadStack();
 
+#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION
```
#if defined CONFIG_IEEE802154_TLX_OPTIMIZATION && CONFIG_IEEE802154_TLX_OPTIMIZATION

haiwentelink commented:
too many commit log ,need to adjust , shoud be 2 commit , one for nvs changes , others should be merged ,then we will review again .

s07641069 reviewed (changes requested):
You made mistake (space) in yaml file and as reason Telink CI not started
check list of Ci and fix
https://github.com/s07641069/connectedhomeip/actions/runs/14100660836

charlie-ai committed:
[telink] adjust thread settings

     - Set the macro CONFIG_ISR_STACK_SIZE
     - Set the macro CONFIG_TL_BLE_CTRL_THREAD_STACK_SIZE
     - set the macro of OPENTHREAD_CSMABACKOFF_OPTIMIZATION

Signed-off-by: Lingzhi Chen <lingzhi.chen@telink-semi.com>

damien0x0023 commented:
Re-open this PR.

charlie-ai committed:
[telink] Solve the problem of shared device running error

     - set the macro of CHIP_TASK_STACK_SIZE
     - set the macro of NET_RX_STACK_SIZE
     - set the macro of NET_TX_STACK_SIZE

Signed-off-by: Lingzhi Chen <lingzhi.chen@telink-semi.com>

olekoz committed:
[Telink] Fix commissioning restart on BLE disconnection

charlie-ai committed:
riscv: telink: test CI.

    - test CI.

Signed-off-by: Lingzhi Chen <lingzhi.chen@telink-semi.com>
